accounts-db/src/accounts_db.rs

```diff
@@ -1210,6 +1226,27 @@ impl AccountStorageEntry {
         self.alive_bytes.load(Ordering::Acquire)
     }
 
+    /// Return true if offset is "new" and inserted successfully. Otherwise,
+    /// return false if the offset exits already.
+    pub fn insert_zero_lamport_single_ref_account_offset(&self, offset: usize) -> bool {
+        let mut zero_lamport_single_ref_offsets =
+            self.zero_lamport_single_ref_offsets.write().unwrap();
+        zero_lamport_single_ref_offsets.insert(offset)
```

Comment:
maybe here we need to check to see if we've crossed the threshold so that we should shrink it. Clean does this.